### PR TITLE
Api tests

### PR DESCRIPTION
**Pull request by agadzhiev**
This PR combine all changes from the Start.
Home work: Add new tests positive and negative roles checks.

### CODE REVIEW

File: src/test/java/com/example/teamcity/api/api/BuildTypeTest.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+        var user2 = generate(User.class);
+        user2.setRoles(generate(Roles.class, "PROJECT_ADMIN", "p:" + project2.getId()));
+        superUserCheckRequests.<User>getRequest(USERS).create(user2);
+
+        testData.getBuildType().setProject(project1);
+        new UncheckedBase(Specifications.authSpec(user2), BUILD_TYPES)
+                .create(testData.getBuildType())
+                .then().assertThat()
+                .statusCode(HttpStatus.SC_FORBIDDEN) // Проверяем, что код ответа 403 Forbidden
+                .body(Matchers.containsString("Access denied. Check the user has enough permissions to perform the operation."));
```
Comment (nobugs-me) [inline src/test/java/com/example/teamcity/api/api/BuildTypeTest.java:87]: В будущем все одинаковые типы ассертов можно вынести в отдельный класс и добавлять набор ассертов как ResponseSpecification

**Review by nobugs-me — APPROVED** [review]
Привет! 

Спасибо за выполнение базового ДЗ! ⚡️

Тест выполнен корректно, поэтому ставлю Approve 🤝
